alt-api-ui/src/schedule-viewer/TutorPicker.tsx

```diff
@@ -0,0 +1,106 @@
+import { faAdd, faCheck, faTrash } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { Button, Input, Modal, Text } from '@nextui-org/react'
+import React, { useState } from 'react'
+import { TutorDatalist } from '../datalists/TutorDatalist'
+
+type TutorPickerProps = {
+  tutors: string[]
+  setTutors: (value: string[]) => void
+  visible: boolean
+  setVisible: (value: React.SetStateAction<boolean>) => void
+}
+
+function shouldDisableButton(input: string, tutors: string[]) {
+  if (!input) return true
+  if (input.length === 0) return true
+  if (tutors.includes(input)) return true
+  return false
+}
+
+export function TutorPicker({
```

Comment:
O ile nie jestem zwolennikiem kopiowania kodu, rozumiem twoją decyzję, gdyż moja implementacja nie przewidywała code reuse. To jest ewidentnie mój błąd, że musiałeś skopiować kod. W kolejnych wersjach poprawię to, gdyż planuję lepszą implementację walidacji danych i kontroli form'a